lib/index.js

```diff
@@ -39,6 +39,164 @@ Wigzo.prototype.loaded = function() {
 };
 
 /**
+ * Product Added
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productAdded = function(track) {
+  var props = track.properties();
+  if (props.product_id) {
+    window.wigzo.track('addtocart', props.product_id);
+  }
+};
+
+
+/**
+ * Wishlist Product Added to Cart
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productAddedFromWishlistToCart = function(track) {
+  var props = track.properties();
+  if (props.product_id) {
+    window.wigzo.track('addtocart', props.product_id);
+  }
+};
+
+/**
+ * Product Removed
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productRemoved = function(track) {
+  var props = track.properties();
+  if (props.product_id) {
+    window.wigzo.track('removedfromcart', props.product_id);
+  }
+};
+
+/**
+ * Products Searched
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productsSearched = function(track) {
+  var props = track.properties();
+  if (props.query) {
+    window.wigzo.track('search', props.query);
+  }
+};
+
+/**
+ * Product Added to Wishlist
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productAddedToWishlist = function(track) {
+  var props = track.properties();
+  if (props.product_id) {
+    window.wigzo.track('wishlist', props.product_id);
+  }
+};
+
+
+/**
+ * Checkout Started
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.checkoutStarted = function(track) {
+  var productList = track.products();
+  var addedProductIds = [];
+  for (var i in productList) {
```

Comment:
refactor this to this for better case handling (also `products` is an array not an object):

```
var productList = track.products();
var addedProductIds = [];
for (var i = 0; i < productList.length; i++) {
   var item = productList[i];
   addedProductIds.push(item.product_id || item.productId);
}

if (addedProductIds.length) window.wigzo.track('checkoutstarted', addedProductIds);
```